clients/drcachesim/scheduler/scheduler.h

```diff
@@ -854,6 +855,16 @@ template <typename RecordType, typename ReaderType> class scheduler_tmpl_t {
         std::unique_ptr<ReaderType> kernel_syscall_reader;
         /** The end reader for #kernel_syscall_reader. */
         std::unique_ptr<ReaderType> kernel_syscall_reader_end;
+        /**
+         * Adds noise generators to the scheduler. A noise generator creates synthetic
+         * trace records that can be scheduled alongside records of one or more real
+         * traces.
+         */
+        bool noise_generator_enable = false;
+        /**
+         * Contains metadata information to drive the noise generation.
+         */
+        std::unique_ptr<noise_generator_info_t> noise_generator_info;
```

Comment:
I'm trying to remember from our offline conversation why the scheduler options need to have noise generator info?  Wouldn't noise generators just be additional input workloads and not need any global flags?  They are part of the scheduler library since scheduler users are the ones who construct them and so we provide them alongside the scheduler proper, but they are modular and separate from the core scheduling logic.  

I.e., it doesn't seem like these two fields need to be here?  They aren't referenced in this PR.

Reply:
Both scheduler options removed.
They are not needed now that we're using a factory: `noise_generator_factory_t`.